tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+        self.registry.register_module(module=self.backbone, force=True)
+        backbone_config = {
+            "type": "MockBackbone",
+        }
+        inputs = {"backbone_config": backbone_config, "registry": self.registry, "backend": "mmseg"}
+        results = update_backbone_args(**inputs)
+        expected_results = ["depth"]
+        assert results == expected_results
+
+    @e2e_pytest_unit
+    def test_update_backbone_args_with_out_indices(self):
+        """Update Args with out_indices."""
+        backbone_config = {"type": "MockBackbone", "out_indices": (0, 1, 2, 3)}
+        self.registry.register_module(module=self.backbone, force=True)
+        inputs = {"backbone_config": backbone_config, "registry": self.registry, "backend": "mmseg"}
+        results = update_backbone_args(**inputs)
+        expected_results = ["depth"]
+        assert results == expected_results
+        assert "use_out_indices" in backbone_config
+
+    @e2e_pytest_unit
+    def test_update_backbone_args_with_option(self):
+        """Update backbone using the backbone name from the backbone list (Check updating with options)."""
+        child_registry = MockRegistry(name="mmseg", parent=self.registry, scope="mmseg")
+        backbone_config = {
+            "type": "mmseg.ResNet",
+        }
+        child_registry.register_module(name="ResNet", module=self.backbone, force=True)
+        inputs = {"backbone_config": backbone_config, "registry": self.registry, "backend": "mmseg"}
+        results = update_backbone_args(**inputs)
+        expected_results = []
+        assert results == expected_results
+        expected_depth = 18
+        assert "depth" in backbone_config
+        assert backbone_config["depth"] == expected_depth
+
+    @e2e_pytest_unit
+    def test_update_backbone_args_without_options(self):
+        """Update backbone using the backbone name from the backbone list (Check updating without options)."""
+
+        def mock_init(self, extra):
+            super(MockBackbone, self).__init__()
+
+        backbone_config = {
+            "type": "mmseg.HRNet",
+        }
+        self.backbone.__init__ = mock_init
+        child_registry = MockRegistry(name="mmseg", parent=self.registry, scope="mmseg")
+        child_registry.register_module(name="HRNet", module=self.backbone, force=True)
+        inputs = {"backbone_config": backbone_config, "registry": self.registry, "backend": "mmseg"}
+        results = update_backbone_args(**inputs)
+        expected_results = ["extra"]
+        assert results == expected_results
+        expected_extra_value = "!!!!!!!!!!!INPUT_HERE!!!!!!!!!!!"
+        assert "extra" in backbone_config
+        assert backbone_config["extra"] == expected_extra_value
+
+    @e2e_pytest_unit
+    def test_update_backbone_args_abnormal_backbone_type(self):
+        """Raise ValueError with unexpected backbone."""
+        backbone_config = {
+            "type": "unexpected",
+        }
```

Comment:
is the line length greater than our limit here?

Reply:
No, I checked and fixed it!